src/main/kotlin/org/elm/ide/typing/ElmQuoteHandler.kt

```diff
@@ -0,0 +1,63 @@
+package org.elm.ide.typing
+
+import com.intellij.codeInsight.editorActions.MultiCharQuoteHandler
+import com.intellij.codeInsight.editorActions.QuoteHandler
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.editor.highlighter.HighlighterIterator
+import org.elm.lang.core.psi.ElmTypes.*
+
+class ElmQuoteHandler : QuoteHandler, MultiCharQuoteHandler {
+    // If this returns true, the editor will move the carat one character right instead of inserting the typed character
```

Comment:
I'm glad you wrote all these comments. It's not obvious at all what's going on just based on the method names.